docs/rfcs/2023-03-29-file-external-table/rfc.md

```diff
@@ -0,0 +1,83 @@
+---
+Feature Name: "File external table"
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/1041
+Date: 2023-03-08
+Author: "Xu Wenkang <wenymedia@gmail.com>"
+---
+
+File external table
+
+---
+
+# Summary
+
+Allows users to perform SQL queries on files
+
+# Motivation
+
+User data may already exist in other storages, i.g., file systems/s3, etc. in CSV, parquet, JSON format, etc. We can provide users the ability to perform SQL queries on these files.
+
+
+# Details
+## Overview
+![overview](external-table-engine-overview.png)
+
+We implement a file table engine that creates an external table by accepting user-specified file paths and treating all files as immutable.
+
+1. File table manifest: contains file paths and connection options.
+2. File table engine: provides the ability to decode files to `RecordBatch` stream.
+
+## Life cycle
+
+### Register table
+1. Writes table metadata to manifest
+2. Creates the table via file table engine
+3. Registers table to catalog
+### Deregister table
+1. Fetches the target table (figure out table engine type)
+2. Deregisters the target table
+3. Finds the target table engine
+4. Drops the target table
+
+```rust 
+    engine_manager.engine("file")?.drop_table("target")
+```
+### Recover
+1. Collects tables name and engine type info
+2. Finds the target tables in different engines
+3. Opens and registers table
+
+## Drawbacks
+- Unsupported filter pushdown
+- Hard to support indexing
+
+# Alternatives
+
+## Using datafusion API
```

Comment:
If I remember it correctly, DataFusion supports "create external table" SQL. We can reuse that.

Reply:
Yes, DataFusion provides this ability, but it still has some problems. One is that we cannot use OpenDAL CacheLayer using the `register_csv`  method because DataFusion implements its own `ObjectStore` If we want to manage the memory usage entirely, we should implement our own `TableProvider` or `Table` .

Reply:
> Yes, DataFusion provides this ability, but it still has some problems. One is that we cannot use OpenDAL CacheLayer using the `register_csv` method because DataFusion implements its own `ObjectStore` If we want to manage the memory usage entirely, we should implement our own `TableProvider` or `Table` .

Mine misunderstood it should be a temporary table🥲